core/store/database.go

```diff
@@ -0,0 +1,22 @@
+package store
+
+// Database provides access to the underlying data base for get set and delete operations of nonconsensus Data
+type DatabaseService interface {
+	GetDataBase() NonConsensusStore
+}
+
+// NonConsensusStore is a simple key-value store that is used to store non-consensus data.
+// Note the non-consensus data is not committed to the blockchain and does not allow iteration
```

Comment:
We can forbid iteration, although, there is no reason to not support it as it's part of the SS interface and all SS backends support it.

Reply:
I was thinking to keep the supported api simple ourselves we can avoid it. Do you think its a requirement for users?

Reply:
Yes, I could see cases where you'd want to iterate over state. E.g. iterate over all WASM contracts or something.

Reply:
the user would still be able to iterate just not natively

Reply:
How? What does "not natively" mean?